### DISCUSSION TITLE
Cannot access camera with the Haishinkit's Camera2Source ( E/libc: Access denied finding property "vendor.camera.aux.packagelist" )

### DISCUSSION THREAD

**Discussion by h-rlpl** [shogo4405/HaishinKit.kt#89] [Q&A] [answered]
Hello Community!
Below are the error codes I'm encountering after HaishinKit example is implemented.

E/libc: Access denied finding property "persist.vendor.camera.privapp.list"
E/libc: Access denied finding property "vendor.camera.aux.packagelist"
E/libc: Access denied finding property "vendor.camera.aux.packagelist"

I allowed camera & audio record permissions. But still no luck in resolving these
What I found is, these errors are occurring when the following lines are executed. If I comment these the app shall run without crashing

`    override fun onResume() {
        super.onResume()
        cameraSource.open(CameraCharacteristics.LENS_FACING_BACK)
    }`

Also this way
`        cameraSource = Camera2Source(requireContext()).apply {
            open(CameraCharacteristics.LENS_FACING_BACK)
        }`

Appreciate your response on this

Update:
It looks like, if camera functinoality implemented inside fragment, the app crashes. Particularly, The app crashing is when the following line is invoked inside a fragment. But, if its invoked in Activity, the camera stream is shown on screen without any issue and no app crash
`override fun onResume() { super.onResume() cameraSource.open(CameraCharacteristics.LENS_FACING_BACK) }`

**Comment by shogo4405** [comment]
- https://stackoverflow.com/questions/55990262/access-denied-for-property-vendor-camera-aux-packagelist
- Did you set `<uses-feature android:name="android.hardware.camera" />`?

**Reply by h-rlpl** [reply]
Yes, I tried it & made sure permissions are allowed. Also tried the following
- made sure no android:background in surfaceview.
- Tried invoking the camerasource.open happen inside the surfaceCreated
Still no luck

**Reply by h-rlpl** [reply]
@shogo4405 It looks like, if camera functinoality implemented inside fragment, the app crashes. Particularly, The app crashing is when the following line is invoked inside a fragment. But, if its invoked in Activity, the camera stream is shown on screen without any issue and no app crash
override fun onResume() { super.onResume() cameraSource.open(CameraCharacteristics.LENS_FACING_BACK) }

**Comment by h-rlpl** [comment, marked as answer]
After further tests, I found that the following lines are the one which is causing the app crash. So I commented these in the MainActivity and now app works as expected without any crash

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            PixelTransformFactory.registerPixelTransform(VkPixelTransform::class)
        }
        
       Camera(Front/Rear) is now accessible. Although the error "E/libc: Access denied finding property "vendor.camera.aux.packagelist" keeps printing on logcat everytime when app launched,  it isnt leading to any app crash now.

**Reply by shogo4405** [reply]
Thank you!!
